hw/snitch_cluster/src/snitch_cluster_wrapper.sv.tpl

```diff
@@ -826,6 +861,58 @@ total_snax_tcdm_ports = total_snax_narrow_ports + total_snax_wide_ports
 
       % endfor
     % endif
+  
+  % elif snax_core_acc[idx_key]['snax_xdma_flag']:
+    
+
+    % for jdx, jdx_key in enumerate(snax_core_acc[idx_key]['snax_acc_dict']):
+
+  // Accelerators controlled with custom instruction format ports
+  ${cfg['name']}_xdma_wrapper # (
+    .tcdm_req_t       ( ${cfg['pkg_name']}::tcdm_req_t ),
+    .tcdm_rsp_t       ( ${cfg['pkg_name']}::tcdm_rsp_t )
+  ) ${jdx_key}  (
+    //-----------------------------
+    // Clock and reset
+    //-----------------------------
+    .clk_i            ( clk_i  ),
+    .rst_ni           ( rst_ni ),
+    //-----------------------------
+    // Cluster Base Address
+    //-----------------------------
+    .cluster_base_addr_i( cluster_base_addr_i ),
+    //-----------------------------
+    // CSR  format control ports
+    //-----------------------------
+    // Request
+    .csr_req_bits_data_i  ( snax_csr_req_data [${idx}] ),
+    .csr_req_bits_addr_i  ( snax_csr_req_addr [${idx}] ),
+    .csr_req_bits_write_i ( snax_csr_req_write[${idx}] ),
+    .csr_req_valid_i ( snax_csr_req_valid[${idx}] ),
+    .csr_req_ready_o ( snax_csr_req_ready[${idx}] ),
+    // Response
+    .csr_rsp_bits_data_o  ( snax_csr_rsp_data [${idx}] ),
+    .csr_rsp_valid_o ( snax_csr_rsp_valid[${idx}] ),
+    .csr_rsp_ready_i ( snax_csr_rsp_ready[${idx}] ),
+    //-----------------------------
+    // Hardware barrier is not supported by xdma at the moment
+    //-----------------------------
+    //-----------------------------
+    // TCDM ports
+    //-----------------------------
+    .tcdm_req_o  ( snax_tcdm_req[${snax_core_acc[idx_key]['snax_acc_dict'][jdx_key]['snax_tcdm_offset_stop']}:${snax_core_acc[idx_key]['snax_acc_dict'][jdx_key]['snax_tcdm_offset_start']}] ),
+    .tcdm_rsp_i  ( snax_tcdm_rsp[${snax_core_acc[idx_key]['snax_acc_dict'][jdx_key]['snax_tcdm_offset_stop']}:${snax_core_acc[idx_key]['snax_acc_dict'][jdx_key]['snax_tcdm_offset_start']}] )
+  );
```

Comment:
Why does it have to be separated from the original accelerator generation? The ports still look the same!

If the point is to re-use the name you can always set the `snax_acc_name` to the xDMA name right?

Reply:
This is our first version, which only supports local loopback. Later this part will go inside snitch cluster and replace the xdma, and should not be generated by mako script but systemverilog's syntax.